app/controllers/simplified-account/settings/service-name/service-name.controller.js

```diff
@@ -0,0 +1,69 @@
+const { body, validationResult } = require('express-validator')
+const { SERVICE_NAME_MAX_LENGTH } = require('../../../../utils/validation/server-side-form-validations')
+const { response } = require('../../../../utils/response')
+const { updateServiceName } = require('../../../../services/service.service')
+const paths = require('../../../../paths')
+const formatSimplifiedAccountPathsFor = require('../../../../utils/simplified-account/format/format-simplified-account-paths-for')
+const formatValidationErrors = require('../../../../utils/simplified-account/format/format-validation-errors')
+
+function get (req, res) {
+  const context = {
+    service_name_en: req.service.serviceName.en,
+    service_name_cy: req.service.serviceName.cy,
+    manage_en: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.serviceName.edit, req.account.service_id, req.account.type),
+    manage_cy: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.serviceName.edit, req.account.service_id, req.account.type) + '?cy=true'
+  }
+  return response(req, res, 'simplified-account/settings/service-name/index', context)
+}
+
+function getEditServiceName (req, res) {
+  const editCy = req.query.cy === 'true'
+  const context = {
+    edit_cy: editCy,
+    back_link: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.serviceName.index, req.account.service_id, req.account.type),
+    submit_link: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.serviceName.edit, req.account.service_id, req.account.type)
+  }
+  if (editCy) {
+    Object.assign(context, { service_name: req.service.serviceName.cy })
+  } else {
+    Object.assign(context, { service_name: req.service.serviceName.en })
+  }
+  return response(req, res, 'simplified-account/settings/service-name/edit-service-name', context)
+}
+
+async function postEditServiceName (req, res) {
+  const editCy = req.body.cy === 'true'
+  const validations = [
+    body('service-name-input').trim().isLength({ max: SERVICE_NAME_MAX_LENGTH }).withMessage(`Service name must be ${SERVICE_NAME_MAX_LENGTH} characters or fewer`)
+  ]
+  // we don't check presence for welsh names
+  if (!editCy) {
+    validations.push(body('service-name-input').trim().notEmpty().withMessage('Service name is required'))
+  }
+
+  await Promise.all(validations.map(validation => validation.run(req)))
+  const errors = validationResult(req)
+  if (!errors.isEmpty()) {
+    const formattedErrors = formatValidationErrors(errors)
+    return response(req, res, 'simplified-account/settings/service-name/edit-service-name', {
```

Comment:
Is this our normal approach?  Responding to a browser `POST` with anything other than a redirect is usually not the best.  If this is how we normally do it then I don't see a reason to change that as part of this PR.

Reply:
I've just taken a quick look and seen at least one other place we're doing this.  I'll add it to my niggle list and there's no reason to update this PR.

Reply:
Yeah this is a common pattern across the base.

We can update it though, I'd like to set some best practice examples for how to build the rest of the settings in this PR for others to follow.

What would you suggest / why is `.render` inappropriate here? 

We aren't changing URL (in the case of form errors) so `.render` felt like the appropriate choice.

Reply:
At this point it's become an established way that I work and I haven't considered the reasons for a while.  I'll look up some stuff about it.

The most direct difference for the user is that if they refresh the page on that page they get a warning in the browser describing the dangers of repeating a `POST` request which isn't actually an issue in this case.

This PR introduces some good things, I don't think we need to introduce the redirect in the same PR (but if we were already using it we should have followed it which is where my initial comment started).

Reply:
The post-redirect-get pattern is common thing, but I wouldn't know how this would work with passing a bunch of contextual info like
```{
      errors: {
        summary: formattedErrors.errorSummary,
        formErrors: formattedErrors.formErrors
      },
      edit_cy: editCy,
      service_name: req.body['service-name-input'],
      back_link: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.serviceName.index, req.account.service_id, req.account.type),
      submit_link: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.serviceName.edit, req.account.service_id, req.account.type)
    }
```
to the view. 
Be good if we have a look at this in the future though.